### PR TITLE
Store output

### PR DESCRIPTION
**Pull request by kelshmo**
This PR stores workflow output to Synapse in a folder or project specified either in the config file as `store_output` or in the `parent_id` variable of `rnaseq_plan`.

`rnaseq_plan` wraps the functions described in `R/functions.R`

- The output objects (or as 'drake' calls them, targets) have static names. Thus, the objects/targets are passed as a character vector in `rnaseq_plan` to [`prepare_results`](https://github.com/Sage-Bionetworks/sageseqr/compare/store-objs?expand=1#diff-1914f62e1613797775065e47b0f506b7c62d4bc06fbb20ffd41707361d433209R596) to write out to temporary files. 

- `store_results` takes the file paths from `prepare_results`, entity names and provenance history (collapsed by `provenance_helper`), and a description to store data to Synapse. Again, wrapped by `rnaseq_plan` where the entity name is also a character vector because it too is static. As the workflow is re-run, we want to enforce or require the same entity name so the Synapse entities are updated appropriately. `forceVersion = FALSE` for `synStore` so that entities are only updated when `md5` has changed.

- `provenance_helper` is unruly. Since explicit versions are optional in the workflow, I had to be kind of verbose to get the list of input Synapse Ids accurate.

@Aryllen I chose you to review since your packages also push data to Synapse. I'm looking forward to any feedback!

### CODE REVIEW

**Comment by kelshmo** [comment]
Here is a `reprex` with a much simplified `plan` (as to not take time running a regression model etc.) that downloads data from Synapse, does something and uploads the file to a public folder syn23630327

``` r
library(sageseqr)
synapser::synLogin()
#> Welcome, Kelsey Montgomery!
#> NULL

modified_plan <- function(metadata_id, metadata_version, counts_id,
                        counts_version, gene_id_input, sample_id_input,
                        factor_input, continuous_input,
                        report_name, parent_id,
                        rownames, config_file, path_to_cache) {
  drake::drake_plan(
    import_metadata = get_data(!!metadata_id,
                               !!metadata_version),
    import_counts = get_data(!!counts_id,
                             !!counts_version),
    counts = tibble::column_to_rownames(import_counts,
                                        var = !!gene_id_input),
    clean_md = clean_covariates(md = import_metadata,
                                factors = !!factor_input,
                                continuous = !!continuous_input,
                                sample_identifier = !!sample_id_input),
    document_inputs = provenance_helper(
      !!metadata_id, !!counts_id,
      !!metadata_version, !!counts_version
    ),
    Synapse = store_results(
      parent_id = !!parent_id,
      targets = list("clean_md"),
      rownames = !!rownames,
      names = list("Covariates"),
      inputs = document_inputs,
      activity_provenance = "Analyze RNA-seq data with sageseqr pkg",
      path_to_cache = !!path_to_cache,
      config_file = !!config_file,
      report_name = !!report_name
    )
  )
}

plan <- modified_plan(
  metadata_id = "syn22090070",
  metadata_version = "2",
  counts_id = "syn22084837",
  counts_version = NULL,
  gene_id_input = "feature",
  sample_id_input = "sampleID",
  factor_input = c("sampleID", "batch", "sex", "diagnosis"),
  continuous_input = c("RIN", "pmi", "ageofdeath"),
  report_name = NULL,
  parent_id = "syn23630327",
  rownames = list(
    "sampleID"
    ),
  config_file = NULL, 
  path_to_cache = "novel-name"
)
drake::make(
  plan,
  cache = drake::new_cache(
    "novel-name"
    )
  )
#> ▶ target import_counts
#> ▶ target import_metadata
#> ▶ target document_inputs
#> ▶ target counts
#> ▶ target clean_md
#> ▶ target Synapse
#> ################################################## Uploading file to Synapse storage ##################################################Uploading [--------------------]0.00%   0.0bytes/290.0bytes  clean_md13bf34ae069a7.tsv     Uploading [####################]100.00%   290.0bytes/290.0bytes (265.5bytes/s) clean_md13bf34ae069a7.tsv Done...
#> Files uploaded to syn23630327
```

<sup>Created on 2020-12-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

File: R/functions.R
Comment (kelshmo) [inline R/functions.R:633]: make rowname `NULL`??

File: R/functions.R
Comment (kelshmo) [inline R/functions.R:692]: Kara suggests breaking store config into it's own function

**Review by Aryllen — APPROVED** [review]
For what it's worth, a quick skim of the code looks fine to me. I have not run your reprex, however. For some reason I cannot install miVC, which makes sageseqr fail to install. I don't have the time at the moment to troubleshoot. I'll approve, but it's understandable if you want someone to actually test that it works before merging.

**Comment by kelshmo** [comment]
I couldn't repo `mvIC` installation error.

**Pull request merged**
